library/lua/utils.lua

```diff
@@ -613,6 +613,55 @@ function processArgs(args, validArgs)
     return result
 end
 
+-- processes commandline options according to optionActions and returns all
+-- argument strings that are not options.
+--
+-- optionActions is a vector with elements in the following format:
+-- {shortOptionName, longOptionAlias, hasArg=boolean, handler=fn}
+-- shortOptionName and handler are required. If the option takes an argument,
+-- it will be passed to the handler function.
+-- longOptionAlias is optional.
+-- hasArgument defaults to false.
+--
+-- example usage:
+--
+-- local filename = nil
+-- local open_readonly = false
+-- local nonoptions = processArgs2(args, {
+--   {'r', handler=function() open_readonly = true end},
+--   {'f', 'filename', hasArg=true,
+--    handler=function(optarg) filename = optarg end}
+-- })
+--
+-- when args is {'first', '-f', 'fname', 'second'}
+-- then filename will be fname and nonoptions will contain {'first', 'second'}
+function processArgs2(args, optionActions)
```

Comment:
Perhaps `processArgsGetopt`, `getopt`, or some variation of `getopt` would be better? If we ever introduce another implementation (however unlikely), I would be hesitant to name it `processArgs3`.

Reply:
hehe true. I'll go with `processArgsGetopt` to highlight the fact that it's an alternate to `processArgs`